File: log/src/main/java/ua/at/tsvetkov/util/interceptor/LogCatInterceptor.java
package ua.at.tsvetkov.util.interceptor;

import android.support.annotation.Nullable;
import android.util.Log;

public class LogCatInterceptor extends LogInterceptor {

   @Override
   public void log(Level level, String tag, String msg, @Nullable Throwable th) {
      switch (level) {
         case VERBOSE: {
            Log.v(tag, msg);
            break;
         }
         case INFO: {
            Log.i(tag, msg);
            break;
         }
         case DEBUG: {
            Log.d(tag, msg);
            break;
         }
         case WARNING: {
            Log.w(tag, msg);
            break;
         }
         case ERROR: {
            Log.e(tag, msg);
            break;
         }
         case WTF: {
            Log.wtf(tag, msg);
            break;
         }
      }
   }

}

File: lib/src/main/java/ua/at/tsvetkov/util/interceptor/LogCatInterceptor.kt
package ua.at.tsvetkov.util.interceptor

import android.util.Log

class LogCatInterceptor : LogInterceptor() {

    override fun log(level: Level, tag: String?, msg: String?, throwable: Throwable?) {
        when (level) {
            Level.VERBOSE -> {
                Log.v(tag, msg ?: "null")
            }
            Level.INFO -> {
                Log.i(tag, msg ?: "null")
            }
            Level.DEBUG -> {
                Log.d(tag, msg ?: "null")
            }
            Level.WARNING -> {
                Log.w(tag, msg ?: "null")
            }
            Level.ERROR -> {
                Log.e(tag, msg ?: "null")
            }
            Level.WTF -> {
                Log.wtf(tag, msg ?: "null")
            }
        }
    }

}